tests/core/v5_1/alexandria/test_subscriptions.py

```diff
@@ -0,0 +1,69 @@
+import pytest
+import trio
+
+from ddht.v5_1.alexandria.messages import PingMessage, PongMessage
+from ddht.v5_1.alexandria.payloads import PingPayload, PongPayload
+
+
+@pytest.mark.trio
+async def test_alexandria_subscription_via_talk_request(
+    alice_alexandria, alice, bob, bob_alexandria
+):
+    async with bob_alexandria.subscribe(PingMessage) as subscription:
+        await alice_alexandria.ping(bob.node_id)
+
+        with trio.fail_after(1):
+            message = await subscription.receive()
+
+        assert isinstance(message.message, PingMessage)
+        assert message.message.payload.enr_seq == alice.enr.sequence_number
+
+
+@pytest.mark.trio
+async def test_alexandria_subscription_via_talk_response(
+    alice_alexandria, alice, bob_alexandria
+):
+    async with bob_alexandria.subscribe(PongMessage) as subscription:
+        with trio.fail_after(1):
+            await bob_alexandria.ping(alice.node_id)
+
+            message = await subscription.receive()
+
+        assert isinstance(message.message, PongMessage)
+        assert message.message.payload.enr_seq == alice.enr.sequence_number
+
+
+@pytest.mark.trio
+async def test_alexandria_subscription_via_talk_request_protocol_mismatch(
+    alice_network, alice, bob, bob_alexandria, autojump_clock
```

Comment:
I'm starting to really enjoy magic.

Reply:
Being able to test timeouts without waiting for things to timeout FTW